### ISSUE TITLE
uploadArchives doesn't produce the artifact for kotlinpoet-classinspector-elements

### ISSUE THREAD

**Issue by Egorand** [square/kotlinpoet#982]
The `publish` job started failing with the following error:

```
* What went wrong:
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Execution failed for task ':kotlinpoet-classinspector-elements:publishMavenPublicationToMavenRepository'.
Use '--warning-mode all' to show the individual deprecation warnings.
> Failed to publish publication 'maven' to repository 'maven'
See https://docs.gradle.org/6.5/userguide/command_line_interface.html#sec:command_line_warnings
   > Artifact kotlinpoet-classinspector-elements-1.7.0-SNAPSHOT.jar wasn't produced by this build.
```

I'm able to repro locally, calling `./gradlew uploadArchives` doesn't trigger tasks that produce the jar. This is a regression introduced by #979, #980 or #981.

**Comment by ZacSweers** [comment]
Ill take this one, pretty sure I know what the issue is

**Comment by Egorand** [comment]
Fixed by #983
